src/iio_wrapper.cpp

```diff
@@ -427,6 +427,136 @@ float IIOWrapper::getTemperature()
+}
+
+float IIOWrapper::getRawAccelerometerY()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_accel_y, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleAccel_y;
+    return result;
+}
+
+float IIOWrapper::getRawAccelerometerZ()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_accel_z, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleAccel_z;
+    return result;
+}
+
+float IIOWrapper::getRawGyroscopeX()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_anglvel_x, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleAngvel_x;
+    return result;
+}
+
+float IIOWrapper::getRawGyroscopeY()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_anglvel_y, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleAngvel_y;
+    return result;
+}
+
+float IIOWrapper::getRawGyroscopeZ()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_anglvel_z, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleAngvel_z;
+    return result;
+}
+
+float IIOWrapper::getRawRotX()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_rot_x, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleRot_x;
+    return result;
+}
+
+float IIOWrapper::getRawRotY()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_rot_y, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleRot_y;
+    return result;
+}
+
+float IIOWrapper::getRawRotZ()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_rot_z, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleRot_z;
+    return result;
+}
+
+float IIOWrapper::getRawVelocityX()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_velocity_x, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleVelocity_x;
+    return result;
+}
+
+float IIOWrapper::getRawVelocityY()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_velocity_y, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleVelocity_y;
+    return result;
+}
+
+float IIOWrapper::getRawVelocityZ()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_velocity_z, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleVelocity_z;
+    return result;
+}
+
+float IIOWrapper::getRawTemperature()
+{
+    long long valueRaw;
+    iio_channel_attr_read_longlong(m_channel_temp, "raw", &valueRaw);
+    float fvalRaw = valueRaw;
+
+    float result = fvalRaw * m_fvalScaleTemp / 1000.0;
+    return result;
+}
+
```

Comment:
these APIs should return the status and the read value should be pass by pointer in the parameter list.
the name is confusing, please change to getRegTemperature, getRegVelocityZ, etc. you are not returning the raw value, you are returning the converted value by performing register reads.